### PR TITLE
Updating mit-open and dagster pipeline settings for warehouse changes

### PR DESCRIPTION
**Pull request by blarghmatey**
- Updated settings for mit-open to accomodate changes made in the structure of the AWS lakehouse
- Updated settings for Dagster pipelines to write to the proper S3 locations after lakehouse layout changes.

-------

#### What are the relevant tickets?
https://github.com/mitodl/open-discussions/issues/3318

### CODE REVIEW

File: pillar/dagster/mitx_enrollments.sls [hunk 1]
```diff
@@ -9,16 +11,10 @@ dagster:
           config:
             s3_bucket: dagster-{{ environment }}
             s3_prefix: pipeline-storage/xpro_edx
-        results_dir:
-          config:
-            outputs_directory_date_format: '%Y%m%d'
-        healthchecks:
```
Comment (shaidar) [inline pillar/dagster/mitx_enrollments.sls:15]: Why are we removing the healthcheck?

Comment (blarghmatey) [reply pillar/dagster/mitx_enrollments.sls:15]: That was never actually set up for this pipeline so removing the unused config. Something to revisit soon though.

**Comment by shaidar** [comment]
👍

**Review by abeglova — APPROVED** [review]

**Comment by shaidar** [comment]
👍

**Pull request merged**